### PR TITLE
Port from EggDebug to the built-in GLib logging framework

### PR DESCRIPTION
**Pull request by rbuj**
Test 1:
```shell
$ G_MESSAGES_DEBUG=PowerManager mate-power-preferences
$ G_MESSAGES_DEBUG=PowerManager mate-power-statistics
```
Test 2:
```shell
$ G_MESSAGES_DEBUG=all mate-power-preferences
$ G_MESSAGES_DEBUG=all mate-power-statistics
```

### CODE REVIEW

File: src/gpm-control.c [hunk 1]
```diff
@@ -269,7 +268,7 @@ gpm_control_suspend (GpmControl *control, GError **error)
 						       NULL,
 						       &dbus_error );
 		if (proxy == NULL) {
-			egg_error("Error connecting to dbus - %s", dbus_error->message);
+			g_error ("Error connecting to dbus - %s", dbus_error->message);
```
Comment (raveit65) [inline src/gpm-control.c]: Build warning!
```
gpm-control.c:272:4: warning: This statement is never executed

                        g_error_free (dbus_error);

                        ^~~~~~~~~~~~
```
https://travis-ci.org/mate-desktop/mate-power-manager/jobs/650313264#L1944

File: src/gpm-control.c [hunk 2]
```diff
@@ -387,7 +386,7 @@ gpm_control_hibernate (GpmControl *control, GError **error)
...
 			g_error_free (dbus_error);
```
Comment (raveit65) [inline src/gpm-control.c]: And another one.
```
gpm-control.c:390:4: warning: This statement is never executed

                        g_error_free (dbus_error);

                        ^~~~~~~~~~~~
```
https://travis-ci.org/mate-desktop/mate-power-manager/jobs/650313264#L1947
Not sure if these warnings are related to PR, but they are still there.

File: src/gpm-manager.c [hunk 3]
```diff
@@ -1773,21 +1772,21 @@ gpm_manager_systemd_inhibit (GDBusProxy *proxy) {
...
+            g_debug("Failed to get systemd inhibitor");
             return r;
         }
-        egg_debug ("System inhibitor fd is %d", fd);
+        g_debug ("System inhibitor fd is %d", fd);
         g_object_unref (fd_list);
         g_variant_unref (res);
     } else if (error != NULL || res == NULL) {
-        egg_error ("Error in dbus - %s", error->message);
+        g_error ("Error in dbus - %s", error->message);
```
Comment (raveit65) [inline src/gpm-manager.c]:
```
gpm-manager.c:1785:40: warning: Access to field 'message' results in a dereference of a null pointer (loaded from variable 'error')

        g_error ("Error in dbus - %s", error->message);

                                       ^~~~~~~~~~~~~~
```
https://travis-ci.org/mate-desktop/mate-power-manager/jobs/650313264#L2356
Not sure if this warning comes from you, but you touched the code :P

**Comment by raveit65** [comment]
All those -Wunused warnings are really a mess when you compare code with build logs as reviewer.
I am really not amused.....

**Comment by raveit65** [comment]
New debug commands are working fine in a quick test.

**Comment by rbuj** [comment]
It's only a `git rebase master`.

test: count warnings by their types
```
$ ./autogen.sh --enable-compile-warnings=maximum --prefix=/usr
$ make &> make.log
$ ./warnings-file.sh make.log 
-Wbad-function-cast 4
-Wdeprecated-declarations 29
-Wmissing-prototypes 6
-Wstrict-prototypes 1
-Wunused-but-set-variable 4
-Wunused-function 3
-Wunused-parameter 161
```
./warnings-file.sh
```sh
FILE=$1
for warning in $(grep warning $FILE | grep -Po '\[\-W.*\]' | sed 's/[][]//g' | sort -u)
do
 type=${warning#"-W"}
 num=$(echo "grep $type $FILE | wc -l" | sh)
 echo "$warning $num"
done
```
There is the same error count in master branch.

**Comment by raveit65** [comment]
Ok, my hope died that a rebase to latest commit at master makes it better.

**Comment by rbuj** [comment]
@raveit65 They are cppcheck warnings. We can try to remove in future.

File: src/gpm-manager.c (hunk 3 above)
Comment (raveit65) [reply src/gpm-manager.c]: PR to fix it. https://github.com/mate-desktop/mate-power-manager/pull/328
I guess 328 needs to merged before this one.

**Comment by raveit65** [comment]
> It's only a git rebase master.

Yes, i know.
But your PR is so big, in result i don't wanna touch it to avoid to break it ;)

**Comment by raveit65** [comment]
```
[rave@mother mate-power-manager]$ git rebase master
First, rewinding head to replay your work on top of it...
Applying: Port from EggDebug to the built-in GLib logging framework
Using index info to reconstruct a base tree...
M	src/gpm-prefs-core.c
.git/rebase-apply/patch:1132: space before tab in indent.
    	  Type <command>G_MESSAGES_DEBUG=PowerManager mate-power-manager --no-daemon</command>,
warning: 1 line adds whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging src/gpm-prefs-core.c
Removing src/egg-debug.h
Removing src/egg-debug.c
Removing applets/inhibit/egg-debug.h
Removing applets/inhibit/egg-debug.c
Removing applets/brightness/egg-debug.h
Removing applets/brightness/egg-debug.c
```
No idea where the `whitespace error` is when rebasing.

**Comment by rbuj** [comment]
@raveit65 The `whitespace error` warning has been fixed.

**Comment by raveit65** [comment]
Conflict after merging https://github.com/mate-desktop/mate-power-manager/commit/9f3d2d0986a6130b1cefc84caae161d75b4f56ba :/

**Comment by rbuj** [comment]
I fixed the merge conflict.

File: src/gpm-control.c (hunk 1 above)
Comment (raveit65) [reply src/gpm-control.c]: Fixed by https://github.com/mate-desktop/mate-power-manager/commit/a270eaa5178246c183ef8e681229b25dc8aff0d2

**Pull request merged**